pkg/runner/run_context.go

```diff
@@ -102,6 +102,14 @@ func (rc *RunContext) startJobContainer() common.Executor {
 	image := rc.platformImage()
 	hostname := rc.hostname()
 
+	username := rc.Config.Secrets["DOCKER_USERNAME"]
+	password := rc.Config.Secrets["DOCKER_PASSWORD"]
+
+	if rc.Run.Job().Container().Credentials["username"] != "" {
```

Comment:
Just reading this again. I bet I should not assume `username` is in the map? I'll fix later.

Reply:
```suggestion
	if len(rc.Run.Job().Container().Credentials) == 2 {
```
this should work

Reply:
But it would be good to check if credentials allow interpolated values like:
`username: ${{ env.USER }}`
(I assume they do)

Reply:
@catthehacker Yes, good call, this is common:

```
username: token
password: ${{ github.token }}
```

Or think `${{ secrets.foo }}`.